apps/playlist/pkg/app/app.go

```diff
@@ -0,0 +1,73 @@
+package app
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-app-sdk/app"
+	"github.com/grafana/grafana-app-sdk/logging"
+	"github.com/grafana/grafana-app-sdk/operator"
+	"github.com/grafana/grafana-app-sdk/resource"
+	"github.com/grafana/grafana-app-sdk/simple"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	playlistv0alpha1 "github.com/grafana/grafana/apps/playlist/pkg/apis/playlist/v0alpha1"
+	"github.com/grafana/grafana/apps/playlist/pkg/watchers"
+)
+
+type PlaylistConfig struct {
+	EnableWatchers bool
+}
+
+func New(cfg app.Config) (app.App, error) {
+	var (
+		playlistWatcher operator.ResourceWatcher
+		err             error
+	)
+
+	playlistConfig, ok := cfg.SpecificConfig.(*PlaylistConfig)
+	if ok && playlistConfig.EnableWatchers {
+		playlistWatcher, err = watchers.NewPlaylistWatcher()
+		if err != nil {
+			return nil, fmt.Errorf("unable to create PlaylistWatcher: %w", err)
+		}
+	}
+
+	simpleConfig := simple.AppConfig{
+		Name:       "playlist",
+		KubeConfig: cfg.KubeConfig,
+		InformerConfig: simple.AppInformerConfig{
+			ErrorHandler: func(ctx context.Context, err error) {
+				logging.DefaultLogger.With("error", err).Error("Informer processing error")
+			},
+		},
+		ManagedKinds: []simple.AppManagedKind{
+			{
+				Kind:    playlistv0alpha1.PlaylistKind(),
+				Watcher: playlistWatcher,
```

Comment:
NIT – Do you think we'd need a watcher over a reconciler? I'm trying to push for using reconciler everywhere as a default option, since it makes a few things easier to work with (specifically caching and handling old vs. new object in updates).

Reply:
this was an outdated diff, and merged in a previous PR, so i am not going to address it here, but for some context: this was the output of codegen, so i went with whatever was generated and modified it. i believe @charandas was planning on adding codegen specifically for grafana services, so this might be something to address there and then we can re-run it for playlists in a follow up PR. sound ok?